modules/ocf_mirrors/files/process-mirrors-logs

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+import argparse
+import os
+from datetime import date
+
+import pymysql
+
+OCFSTATS_PWD = os.environ['OCFSTATS_PWD']
+MIRRORS_PATH = '/opt/mirrors/ftp'
+LOG_PATH = '/var/log/apache2'
+LOG_NAME = 'mirrors.ocf.berkeley.edu_access.log'
+
+sources = [mirrored for mirrored in os.listdir(MIRRORS_PATH)
+           if os.path.isdir(os.path.join(MIRRORS_PATH, mirrored)) and not mirrored.startswith('.')]
+
+sources.append('other')  # catchall
+
+
+def build_dists():
+    return {mirrored: {'up': 0, 'down': 0} for mirrored in sources}
+
+
+def process_file(fn):
+    dists = build_dists()
+    with open(fn, 'r') as f:
+        for line in f:
+            stats = line.split()
+
+            # extract dist name from request url
+            # '/debian/pool/main/h/hwdata/...' -> 'debian'
+            dist = stats[6]
```

Comment:
I wrote this script as a bit of a hack to quickly see what traffic levels we had across dists, but ideally there would be a comment here explaining why these indices were chosen with maybe an example line:

The format comes from [here](https://github.com/ocf/puppet/blob/2e9e2cd12819653c7eb4c8ad362736b4f65122aa/modules/ocf_mirrors/manifests/init.pp#L56):

    %h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\" %I %O

which translates to something like

    8.8.8.8 - - [26/Jun/2017:00:05:13 -0700] "GET /centos/7.3.1611/path/here/blah HTTP/1.1" 200 3735 "-" "urlgrabber/3.10 yum/3.4.3" 147 3971

so that's why the 6th index is needed (`%r` corresponds to the path because there is an extra space in the timestamp), and the 8th index below gives the status code so that can be filtered for only successes. -1 (`%O`) and -2 (`%I`) below give the bytes downloaded and uploaded as seen by Apache respectively.